ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -565,9 +368,9 @@ public void lockAndRespond() throws CommandProcessorException {
           "No previously compiled query for driver - queryId=" + driverContext.getQueryState().getQueryId());
     }
 
-    if (requiresLock()) {
```

Comment:
What about driverTxnHandler.acquireLocksIfNeeded, and pushing the logic to the TxnHandler?

Reply:
I agree, I'll modify this.